src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/BuildPropertyPage/DefineConstantsVBValueProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties;
+
+/// <summary>
+/// Supports the setting of constants in the property page, corresponding to the "DefineConstants" property
+/// </summary>
+[ExportInterceptingPropertyValueProvider(DefineConstantsPropertyName, ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+[AppliesTo(ProjectCapability.VisualBasic)]
+internal sealed class DefineConstantsVBValueProvider : InterceptingPropertyValueProviderBase
+{
+    internal const string DefineConstantsPropertyName = "DefineConstants";
+
+    public override Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+    {
+        Dictionary<string, string> constantsDictionary = StringListEncoding.ParseIntoDictionary(unevaluatedPropertyValue);
+        return Task.FromResult<string?>(KeyQuotedValuePairListEncoding.Instance.Format(StringListEncoding.EnumerateDictionary(constantsDictionary)));
```

Comment:
As mentioned yesterday, looks nicer as a private field instead of singleton, but up to you